mountpoint-s3-client/src/object_client.rs

```diff
@@ -143,7 +143,7 @@ pub trait ObjectClient {
 ///
 /// [`ServiceError`]: ObjectClientError::ServiceError
 /// [`ClientError`]: ObjectClientError::ClientError
-#[derive(Debug, Error)]
+#[derive(Debug, Error, PartialEq)]
```

Comment:
Is this required?

Reply:
Yes, otherwise `assert_eq!(get_request.get_object_metadata().await, Ok(object_metadata));` fails in the tests. I could add a `.unwrap` but that doesn't feel right.